internal/provider/org/group_resource.go

```diff
@@ -0,0 +1,309 @@
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.CloudAvenue)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *client.CloudAvenue, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *orgGroupResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Retrieve values from plan
+	var (
+		plan *orgResourceModel
+		err  error
+	)
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// catalog creation is accessible only for administator account
+	adminOrg, err := r.client.Vmware.GetAdminOrgByNameOrId(r.client.GetOrg())
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving Org", err.Error())
+		return
+	}
+
+	// Get role reference
+	roleRef, err := adminOrg.GetRoleReference(plan.Role.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving role reference", err.Error())
+		return
+	}
+
+	// Create the org group
+	newGroup := govcd.NewGroup(&r.client.Vmware.Client, adminOrg)
+	groupDefinition := govcdtypes.Group{
+		Name:         plan.Name.ValueString(),
+		Role:         roleRef,
+		ProviderType: "SAML",
+		Description:  plan.Description.ValueString(),
+	}
+
+	newGroup.Group = &groupDefinition
+
+	createGroup, err := adminOrg.CreateGroup(newGroup.Group)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating Org group", err.Error())
+		return
+	}
+
+	var userNames []attr.Value
+
+	plan = &orgResourceModel{
+		ID:          types.StringValue(createGroup.Group.ID),
+		Name:        plan.Name,
+		Description: plan.Description,
+		Role:        plan.Role,
+		UserNames:   types.ListValueMust(types.StringType, userNames),
+	}
+
+	// Set state to fully populated data
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *orgGroupResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state *orgResourceModel
+
+	// Get current state
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	adminOrg, err := r.client.Vmware.GetAdminOrgByNameOrId(r.client.GetOrg())
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving Org", err.Error())
+		return
+	}
+
+	group, err := adminOrg.GetGroupById(state.ID.ValueString(), false)
+	if govcd.IsNotFound(err) {
+		// Group not found, remove from state
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving Org group", err.Error())
+		return
+	}
```

Comment:
Prevent `nil` panic on `govcd.IsNotFound`

```suggestion
	group, err := adminOrg.GetGroupById(state.ID.ValueString(), false)
	if err != nil {
		if govcd.IsNotFound(err) {
			// Group not found, remove from state
			resp.State.RemoveResource(ctx)
			return
	        }
		resp.Diagnostics.AddError("Error retrieving Org group", err.Error())
		return
	}
```